### ISSUE TITLE
Patch testbed agents before swap of Python-based to Go-based DBS Server is performed

### ISSUE THREAD

**Issue by amaltaro** [dmwm/WMCore#10882]
**Impact of the new feature**
Testbed WMAgents

**Is your feature request related to a problem? Please describe.**
As defined in the document for the DBS2Go migration, once the Go-based DBS Server and the stakeholders applications are ready (meaning, they are already using the compatible dbs3-client library), we plan to swap the cmsweb-testbed backend from the Python-based to the Go-based DBS Server. In other words, the endpoint and REST URI/URLs will be the same, but requests will be processed and served by the new Go server.

**Describe the solution you'd like**
Our latest stable WMAgent release is `1.5.4.patch3`, which uses dbs3-client version `3.16.2001` from the **old** DBS repository. So we need to either:
a) collect all the patches that went in into DBSClient repository, to provide the compatibility and aggregation layer, and port those to the DBS repo such that we can create a patch to be applied on the agents.
b) or, we build a new release (1.5.4.patch4 ?) with the new dbs3-client, drain those testbed agents and redeploy them. It should be fairly easy, but still we need to give a few days notice to people using those agents.

In short, actions to be taken with this ticket are:
* come up with a dbs3-client solution that will allow testbed agents to talk to either DBS Server transparently
* patch the testbed agents that are currently used for Opportunistic resources exploration (CMS@Home, BSC, M100) 

**Describe alternatives you've considered**
Alternative given above (either patch creation; or new release)

**Additional context**
This is only required if - by the time we are ready to switch cmsweb-testbed backends - we are still running WMAgents on 1.5.4 series. 

Might depend on https://github.com/dmwm/WMCore/issues/10884

**Comment by amaltaro** [comment]
This switch has already happened in cmsweb-testbed and I haven't heard of any issues affecting the testbed agents. Still to be verified in the logs of the agents though.

**Comment by amaltaro** [comment]
WMAgent `1.5.7.patch1` stable release has been made available and developers are invited to upgrade their agents.
I started upgrading testbed agents, and I am in contact with Federica/Ivan to upgrade the CMS@Home agent as well.

I consider this issue no longer relevant. Closing.
